arangod/Aql/IndexNodeOptimizerRules.cpp

```diff
@@ -118,6 +119,14 @@ void arangodb::aql::lateDocumentMaterializationRule(Optimizer* opt,
       if (!indexNode->canApplyLateDocumentMaterializationRule() || indexNode->isLateMaterialized()) {
         continue; // loop is already optimized
       }
+      auto producingNode = dynamic_cast<DocumentProducingNode const*>(loop);
```

Comment:
Why do you use a cast here? You can check indexNode. It is better to use ExecutionNode::castTo<>().

Reply:
true